networklibrary/src/main/java/com/eis/smsnetwork/commands/AddResource.java

```diff
@@ -0,0 +1,39 @@
+package com.eis.smsnetwork.commands;
```

Comment:
Why are these commands restricted to SMS? Couldn't we have
```Java
public class AddResource<D extends NetDictionary, B extends NetBroadcaster> implements Command{}

public class com.eis.smsnetwork.commands.AddResource extends com.eis.communication.network.AddResource<SMSNetDictionary, SMSNetBroadcaster>{}
```
so that every implementation can use the same command class.

Also, there is no need for the second class (that would become empty, just a placeholder), inside `NetworkDictionary` you could use `CommandExecutor(new AddResource<SMSNetDictionary, SMSNetBroadcaster>(dict, broadcaster))`

Reply:
wait, why in NetworkDictionary?

Reply:
I don't know, I thought that's where these commands will be created, am I wrong?

Reply:
Also yeah, we could totally have generics instead of restricting to SMS messages, but I'd also like to have something working before doing so.

Reply:
Using generic would make it still "working". Also now that I have had some more time to think, there is no need for generics, you could pass an interface of `NetDictionary` as a parameter and call `dict.addResource()` it would still work

Reply:
```Java
public class AddResource implements Command {

    private String key;
    private String value;
    private NetDictionary netDictionary;

    /**
     * Constructor for the AddResource command, needs the data to operate
     * @param key The key of the resource to add
     * @param value The value of the resource to add
     * @param netDictionary The dictionary to add the resource in
     */
    public AddResource(@NonNull String key,@NonNull String value,@NonNull NetDictionary netDictionary){
        this.key = key;
        this.value = value;
        this.netDictionary = netDictionary;
    }

    /**
     * Adds the key-resource pair to the dictionary, then broadcasts the message
     */
    public void execute() {
        netDictionary.addResource(key, value);
        //TODO broadcast
    }
}
```

This does the same exact job and is not restricted to SMS

Reply:
the network dictionary just keeps the resources, it's just a data structure, it shouldn't do any work outside of that.
Commands get created where requests starts, as in methods of NetworkManager
(`invite`, `addPeer`, ...)

Reply:
Oh fuck me, yes, I confused Dictionary and Manager. I meant `NetworkManager` sorry.

Reply:
Anyway, as I said, we can totally do that! But I'd like to have something working before doing so.
The deadline is getting closer afterall

Reply:
Are you aware that applying the changes I proposed is going to let the class exactly as it is right now, do you? It's going to work the same way but with widened pre-conditions.

Reply:
Oh yeah you're right.
Oke, I'll update it. 👍

Reply:
It's just a matter of removing SMS- prefix and moving it inside another package, everything else stays the same (except a renaming action)